src/main/java/com/github/minecraftschurlimods/arsmagicalegacy/client/AMRenderTypes.java

```diff
@@ -1,38 +1,90 @@
+                        AMShaders.setUniform("center", state.getCenterX(), state.getCenterY());
+                        AMShaders.setUniform("radius", state.getRadius());
+                        AMShaders.setUniform("brightness", state.getBrightness());
                     }, () -> {}))
                     .createCompositeState(false)
     );
+    public static final RenderType BEAM_CORE = RenderType.create(
+            "beam_core",
+            DefaultVertexFormat.POSITION_COLOR_TEX,
+            VertexFormat.Mode.QUADS,
+            256,
+            false,
+            false,
+            RenderType.CompositeState.builder()
+                    .setTextureState(BEAM_CORE_TEXTURE)
+                    .setShaderState(RenderStateShard.ShaderStateShard.POSITION_COLOR_TEX_SHADER)
+                    .setLayeringState(VIEW_OFFSET_Z_LAYERING)
+                    .setTransparencyState(TRANSLUCENT_TRANSPARENCY)
+                    .setDepthTestState(NO_DEPTH_TEST)
+                    .setCullState(NO_CULL)
+                    .setLightmapState(NO_LIGHTMAP)
+                    .setWriteMaskState(COLOR_WRITE)
+                    .createCompositeState(false));
+    public static final RenderType BEAM_MAIN = create(
+            "beam_main",
+            DefaultVertexFormat.POSITION_COLOR_TEX,
+            VertexFormat.Mode.QUADS,
+            256,
+            false,
+            false,
+            RenderType.CompositeState.builder()
+                    .setTextureState(BEAM_MAIN_TEXTURE)
+                    .setShaderState(RenderStateShard.ShaderStateShard.POSITION_COLOR_TEX_SHADER)
+                    .setLayeringState(VIEW_OFFSET_Z_LAYERING)
+                    .setTransparencyState(TRANSLUCENT_TRANSPARENCY)
+                    .setDepthTestState(NO_DEPTH_TEST)
+                    .setCullState(NO_CULL)
+                    .setLightmapState(NO_LIGHTMAP)
+                    .setWriteMaskState(COLOR_WRITE)
+                    .createCompositeState(false));
+    public static final RenderType BEAM_GLOW = create("beam_glow",
+            DefaultVertexFormat.POSITION_COLOR_TEX,
+            VertexFormat.Mode.QUADS,
+            256,
+            false,
+            false,
+            RenderType.CompositeState.builder()
+                    .setTextureState(BEAM_GLOW_TEXTURE)
+                    .setShaderState(RenderStateShard.ShaderStateShard.POSITION_COLOR_TEX_SHADER)
+                    .setLayeringState(VIEW_OFFSET_Z_LAYERING)
+                    .setTransparencyState(TRANSLUCENT_TRANSPARENCY)
+                    .setDepthTestState(NO_DEPTH_TEST)
+                    .setCullState(NO_CULL)
+                    .setLightmapState(NO_LIGHTMAP)
+                    .setWriteMaskState(COLOR_WRITE)
+                    .createCompositeState(false));
+
+    // We need to extend RenderType for access to the protected fields, but we should never actually instantiate this class.
```

Comment:
please use an AT instead

Reply:
I really don't want to shotgun AT here (this affects several fields)

Reply:
still this is worse